sky/spot/spot_state.py

```diff
@@ -344,7 +358,7 @@ def set_failed(job_id: int,
                failure_type: SpotStatus,
                failure_reason: str,
                end_time: Optional[float] = None):
-    """Set the job to failed.
+    """Set the task to failed, if they are in non-terminal states.
```

Comment:
Well here task_id is Optional, so we may want to say "...an entire job or a task, if they are ..."